src/components/modals/ActionCreatorEditor/ActionCreatorEditor.tsx

```diff
@@ -293,7 +296,8 @@ interface ComponentState {
     slateValuesMap: SlateValueMap
     secondarySlateValuesMap: SlateValueMap
     isTerminal: boolean
-    reprompt: boolean
+    shouldReprompt: boolean
```

Comment:
It's good that you renamed the variable to better suite a boolean, but I would rather it follow the pattern of the other boolean variables with `is` prefix. Should seems better suited when it's followed by a verb when then implies the action object is doing the reprompting, but it's actually the server. The action simply has passive properties that are observed.  It's a similar reason for `isTerminal` rather than `shouldRescore`.

Reply:
Not sure I understand your comment.  "isReprompt" seems misleading as the action isn't a reprompt action.  It's stating whether the action should reprompt if it gets an unexpected input

Reply:
> "isReprompt" seems misleading as the action isn't a reprompt action.

I'm not suggesting "isReprompt" should be used. I'm saying the problem is trying to make a verb style variable into a boolean by adding "should".  It should be a noun. Verbs are for methods not properties.

> It's stating whether the action should reprompt if it gets an unexpected input

This is the problem.  In your mind, you think the action is doing the reprompting.  In my mind this is not the correct way to describe this behavior because it bakes in all the assumptions about unrelated things. The focus should be on why it repromts, and the effect of detecting unexpected input. The result of unexpected input could be anything but we previously chose to reprompt.  Now, we're saying it could be any action (even an action that is not a prompt) which means it's not reprompting anymore. It could be to escalate to human, or to start the conversation over, or something else like "lets try that again".

Names like `hasUnexpectedInputAction` seems more appropriate.

Reply:
As the field on ActionBase is called "repromptActionId", it think it's less confusing to have them aligned